app/models/course_lesson.rb

```diff
@@ -14,11 +14,26 @@ class CourseLesson < ApplicationRecord
   has_many :course_lesson_parts
 
   validates :title, presence: { message: "Enter a title" }, length: { maximum: 255 }
+  validates :completion_time_in_minutes, numericality: { greater_than_or_equal_to: 1, allow_nil: true, message: "Enter a number greater than 0" }
 
   attr_accessor :progress
 
   def course_lesson_parts_in_order
     preloaded_parts = course_lesson_parts.includes(:previous_lesson_part, :next_lesson_part)
     elements_in_order(elements: preloaded_parts, previous_method_name: :previous_lesson_part)
   end
+
+  def convert_minutes_to_words(minutes_in_time)
+    number_of_hours = minutes_in_time / 60
+    number_of_minutes = minutes_in_time % 60
+
+    hour = "hour".pluralize(number_of_hours)
+    minute = "minute".pluralize(number_of_minutes)
+
+    if number_of_hours.positive?
+      "#{number_of_hours} #{hour} #{number_of_minutes} #{minute}"
+    else
+      "#{number_of_minutes} #{minute}"
+    end
+  end
```

Comment:
Do you mean instead of as in the pic above it'd be "one hour fifty-nine minutes"? Lines 30 and 31 probably aren't clear enough in that their setting a singular or plural string.

Reply:
No, I mean instead of method
```
  def convert_minutes_to_words(minutes_in_time)
```
I want
```
  def duration_in_minutes_in_words
```

As is now, any random object could ask lesson to convert some minutes into human readable format - and that's not lesson's job. Lesson's job is to display its own time in a nice format.

Reply:
Also, I don't want the method to take an argument - cause that's what makes it into a tool that anyone can use. 

I want it to take number of minutes from the field on lesson object - so that the method has a narrower purpose.

Reply:
Ok, my bad, I knew you could access like it relationships like you've used course_lesson_parts but didn't try to just use a key. Seems obvious but at the same time, how does it do that for each one????

Reply:
Haha, Rails magic is the answer. 

By convention any field from the database can be accessed with a key, sorry I didn't make it more obvious, it is not at all clear if you don't expect it.

The underlying mechanism is, uhhhh, more complex. When a Ruby object receives a method call, it tries finding that method in its definitions, and when it doesn't, it calls a special 'method not found' method. Also, all classes are editable at runtime.

I'm not sure which one of these tools Rails uses - it might be adding field methods to models based on schema at runtime, it could be catching calls to fields methods and using its clever machinery to produce the result. Scary to think about it.